### PR TITLE
Fixing password-reset 

### PR DESCRIPTION
**Pull request by Parastoo-S**
The list of programs are not updated for when users want to update their info

### CODE REVIEW

**Comment by Parastoo-S** [comment]
I think the issue with the password-reset should be resolved now

**Comment by Parastoo-S** [comment]
The select2 (dropdown for programs) is not working. Still trying to fix it.

File: db/migrate/20170809153751_update_old_users_required_fields.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class UpdateOldUsersRequiredFields < ActiveRecord::Migration
+  def change
+    User.where(name: nil).update_all(name: "anonymous")
+    User.where(gender: nil).update_all(gender: "unknown")
```
Comment (tahnok) [inline db/migrate/20170809153751_update_old_users_required_fields.rb]: It might make sense to go even further with this and move all users that don't fit in one of our current bucks into 'unknown'. That should probably be a different PR though

File: db/schema.rb [hunk 2]
```diff
@@ -11,7 +11,9 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20170717220935) do
+
+
+ActiveRecord::Schema.define(version: 20170809153751) do
```
Comment (tahnok) [inline db/schema.rb]: This looks weird?

**Comment by Parastoo-S** [comment]
@tahnok At the moment even invalid users can change their password because of the migrations that updates the required fields of name, gender, and identity. 
So those fields are filled with some values that make the user valid.
but if that's not enough I'll remove the check for validations for password_reset.

File: db/schema.rb [hunk 3]
```diff
@@ -11,7 +11,10 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20170808162101) do
+
+
+
+ActiveRecord::Schema.define(version: 20170809153751) do
```
Comment (tahnok) [inline db/schema.rb]: Looks like you need to re-dump the schema here. password/share_type

File: test/controllers/users_controller_test.rb [hunk 4]
```diff
@@ -99,4 +99,41 @@ class UsersControllerTest < ActionController::TestCase
...
+    assert_redirected_to settings_profile_path
+  end
+
+  test "old valid user should be able to update profile with valid inputs" do
+    session[:user_id] = User.find_by(username: "john").id
+    session[:expires_at] = "Sat, 03 Jun 2020 05:01:41 UTC +00:00"
+
+    patch :update, username: "john", user: {name: "John", gender: "Male", identity: "grad", student_id: 9876543, faculty: "arts", program: "Honours BA in English", year_of_study: "3"}
+    assert_equal 'Profile updated successfully.', flash[:notice]
+    assert_equal "John", User.find_by(username: "john").name
```
Comment (tahnok) [inline test/controllers/users_controller_test.rb]: Why not use a variable for User here?

File: db/schema.rb (hunk 3 above)
Comment (Parastoo-S) [reply db/schema.rb]: done

File: test/controllers/users_controller_test.rb (hunk 4 above)
Comment (Parastoo-S) [reply test/controllers/users_controller_test.rb]: modified
